scripts/WSL.ps1

```diff
@@ -1,30 +1,39 @@
-choco install -y Microsoft-Windows-Subsystem-Linux --source="'windowsfeatures'"
+if (
```

Comment:
Better to have this if statement be one that returns if it needs to (because WSL is not supported) instead of wrapping the rest of the code.

Reply:
1. Yep, this permits FCU and newer Windows Server builds to install WSL
2. That certainly seems cleaner; could also allow use elsewhere for possible future dependencies on WSL. I'll do that

Reply:
(I just looked at the wsl install script, I hope I'm commenting in the right place...trying to be helpful.)

There are some gotchas on the installing the various distros, not all have a simple install option and return to the command prompt.

You can look at my Chocolatey package install scripts for examples.
https://chocolatey.org/packages/wsl-ubuntu-1804
https://chocolatey.org/packages/wsl-ubuntu-1604
https://chocolatey.org/packages/wsl-debiangnulinux
https://chocolatey.org/packages/wsl-kalilinux
https://chocolatey.org/packages/wsl-opensuse
https://chocolatey.org/packages/wsl-archlinux
https://chocolatey.org/packages/wsl-sles

Maybe distros would be better handled by choco install commands? (Yes, I'm open to improvements in my install scripts.)

Reply:
(Cross posted from https://github.com/Microsoft/windows-dev-box-setup-scripts/issues/23#issuecomment-440888813)

I mention this because what I saw in the commit 969035c for SLES and openSUSE won't work based on my experience.

openSUSE REF: https://github.com/bcurran3/ChocolateyPackages/blob/master/wsl-opensuse/tools/ChocolateyInstall.ps1
SLES REF: https://github.com/bcurran3/ChocolateyPackages/blob/master/wsl-sles/tools/ChocolateyInstall.ps1

Reply:
Also note that WSL requires a reboot before use/installing distros.

Reply:
Thanks @bcurran3.  I agree the choco install commands make it easier to install the distro.  Thank you for taking initiative in making these.  I'd like the distro owners to review your choco packages before we add them to this project.  I'll discuss with @tara-raj and we can start going through the list of distro owners.

A challenge with installing a WSL distro as part of a boxstarter script is the default user creation so we can also automate the install of tools inside the distro.  In the case of Ubuntu we use the default root user with empty password, allowing us to then automate install of packages.  After the boxstarter script completes the user is encouraged to then create a new user in Ubuntu with a non-blank password.  I see you do the same thing.  As long as the user remembers to go in and create a user with a non-blank password and use that as the default then they're back in the same setup as when installing from the Store.  It may be possible to improve on this so a choco install and Store install produce the same first run experience - this is a topic I'd love to hear feedback on from the community and distro owners.

Reply:
moving the above comment to issue #32

Reply:
@yodurr I'm game for improvements. My scripts for the installs are not the best, but the best I could do. I welcome feedback/changes/PRs. :)